Tardigrade reference architecture and documentation (#6)
plus3it/tardigrade: docs/_pages/structure.md

---
title: "Structure"
permalink: /structure/
toc: true
---

# File Structure
```
tardigrade/

> ├── aws                                    # AWS parition

Reviewer: ```suggestion
├── aws                                    # AWS partition
```

│   ├── 609421024627                       # idempotental element to describe the account
│   │   ├── base                           # baseline configuration
│   │   │   ├── tenant-local.auto.tfvars   # defined variables valuables specific to the account
│   │   │   ├── terragrunt.hcl             # terragrunt configuration for the account
│   │   └── custom                         # custom configuration
│   │       ├── terragrunt.hcl             # terragrunt configuration for the account
│   │       └── variables.tf               # variable declaration specific to the customization
│   ├── tenant.tf                          # partition provider and variable declarations
│   ├── tenant-global.tfvars.yaml          # partition variable values
│   └── terragrunt.hcl                     # terragrunt configuration

> └── baseline                               # account baseline

Reviewer: this should probably be under "roots/" or "modules/" or similar, as there could be multiple baseline configs handling tf logic.

```

## Partition

> In this example, the partition is aws (i.e., `tardigrade/aws`). Generally speaking, most AWS implementations of tardigrade  will only use the `aws` partition. The `aws` parition is the commercial partition that most people are used to. There are other paritions though, such as AWS GovCloud (`aws-us-gov`) and AWS China (`aws-cn`)

Reviewer: ```suggestion
In this example, the partition is aws (i.e., `tardigrade/aws`). Generally speaking, most AWS implementations of tardigrade  will only use the `aws` partition. The `aws` partition is the commercial partition that most people are used to. There are other partitions though, such as AWS GovCloud (`aws-us-gov`) and AWS China (`aws-cn`)
```

### Idempotent Element

> Each directory at this level (e.g., `tardigrade/aws/<accounts>`) represents an individual account. For AWS, the account number is an idempotenet value so we used that to represent individual accounts. The names of these directories do not influence the operation of this framework in any way. They simply serve as a mechanism for developers to easily identify the account.

Reviewer: ```suggestion
Each directory at this level (e.g., `tardigrade/aws/<accounts>`) represents an individual account. For AWS, the account number is an idempotent value so we used that to represent individual accounts. The names of these directories do not influence the operation of this framework in any way. They simply serve as a mechanism for developers to easily identify the account.
```
